app/assets/javascripts/discourse/app/lib/desktop-notifications.js

```diff
@@ -136,31 +136,48 @@ function setupNotifications(appEvents) {
   appEvents.on("page:changed", resetIdle);
 }
 
-export function resetIdle() {
+function resetIdle() {
   lastAction = Date.now();
 }
+
 function isIdle() {
   return lastAction + idleThresholdTime < Date.now();
 }
 
-// Call-in point from message bus
-async function onNotification(data, siteSettings, user, appEvents) {
-  if (!liveEnabled) {
-    return;
-  }
+function setLastAction(time) {
+  lastAction = time;
+}
+
+function canUserReceiveNotifications(user) {
   if (!primaryTab) {
-    return;
+    return false;
   }
+
   if (!isIdle()) {
-    return;
+    return false;
   }
+
   if (user.isInDoNotDisturb()) {
-    return;
+    return false;
   }
+
   if (keyValueStore.getItem("notifications-disabled")) {
+    return false;
+  }
+
+  return true;
+}
+
+// Call-in point from message bus
+async function onNotification(data, siteSettings, user, appEvents) {
+  if (!canUserReceiveNotifications(user)) {
     return;
```

Comment:
Nitpick, but should we `return false;` here as well instead of returning `undefined`?

Reply:
yes would be slightly better I guess